core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java

```diff
@@ -121,10 +119,10 @@ public static Slice currentTimeZone(ConnectorSession session)
     }
 
     @ScalarFunction("from_unixtime")
-    @SqlType("timestamp(3)")
-    public static long fromUnixTime(@SqlType(StandardTypes.DOUBLE) double unixTime)
+    @SqlType("timestamp(3) with time zone")
+    public static long fromUnixTime(ConnectorSession session, @SqlType(StandardTypes.DOUBLE) double unixTime)
     {
-        return Math.round(unixTime * MICROSECONDS_PER_SECOND);
+        return packDateTimeWithZone(Math.round(unixTime * 1000), session.getTimeZoneKey());
```

Comment:
Some points in time are non-representable in the session zone.
maybe add
```
// TODO (https://github.com/trinodb/trino/issues/5781) ....
```


here

(it doesn't make sense to fix this problem for this method, but not fix it elsewhere, hence TODO only)

(https://github.com/trinodb/trino/issues/5781)